vita3k/kernel/src/load_self.cpp

```diff
@@ -70,13 +70,19 @@ static bool load_var_imports(const uint32_t *nids, const Ptr<uint32_t> *entries,
 
         VarImportsHeader *const var_reloc_header = reinterpret_cast<VarImportsHeader *>(entry.get(mem));
         const auto var_reloc_entries = static_cast<void *>(var_reloc_header + 1);
+        const uint32_t reloc_size = (var_reloc_header->reloc_data_size > sizeof(VarImportsHeader)) ? (var_reloc_header->reloc_data_size - sizeof(VarImportsHeader)) : 0;
 
         Address export_address;
         const ExportNids::iterator export_address_it = kernel.export_nids.find(nid);
```

Comment:
This was already a fault in the code but can you fix it? the call to `export_nids.find` should be with a shared_lock (it is not thread safe).